src/pmem.c

```diff
@@ -60,34 +59,43 @@ void pmemThresholdInit(void)
             serverAssert(NULL);
     }
 }
-//TODO: modify logic to check the trend of actual Ratio using pmem_checkpoint_value and dram_checkpoint_value
+
 void adjustPmemThresholdCycle(void) {
+    double step;
     if (server.memory_alloc_policy == MEM_POLICY_RATIO) {
         run_with_period(server.ratio_check_period) {
+            step = 0.05;
             size_t pmem_memory = zmalloc_used_pmem_memory();
             size_t dram_memory = zmalloc_used_memory();
-            size_t total_memory = pmem_memory + dram_memory;
-            size_t total_memory_checkpoint = server.pmem_checkpoint_value + server.dram_checkpoint_value;
+            size_t total_memory_actual = pmem_memory + dram_memory;
             // do not modify threshold when change in memory usage is too small
-            if (absDiff(total_memory_checkpoint, total_memory) > 100) {
+            if (absDiff(server.total_memory_checkpoint, total_memory_actual) > 100) {
+
                 //revert logic to avoid division by zero
-                double setting_state = (double)server.dram_pmem_ratio.pmem_val/server.dram_pmem_ratio.dram_val;
-                double current_state = (double)pmem_memory/dram_memory;
-                size_t threshold = zmalloc_get_threshold();
-                if (fabs(setting_state-current_state) > 0.1) {
-                    if (setting_state < current_state) {
-                        size_t higher_threshold = THRESHOLD_UP(threshold);
-                        if (higher_threshold > server.dynamic_threshold_max) return;
-                        zmalloc_set_threshold(higher_threshold);
+                double target_ratio = (double)server.dram_pmem_ratio.pmem_val/server.dram_pmem_ratio.dram_val;
```

Comment:
The target ratio should be a const value, we don't need to calcuate everytime here. consider to put in  a global place. I prefer we can have the target_ratio in the server structure